java/vortex-jni/src/main/java/dev/vortex/jni/JNIArray.java

```diff
@@ -16,17 +16,25 @@
 package dev.vortex.jni;
 
 import com.google.common.base.Preconditions;
-import com.jakewharton.nopen.annotation.Open;
 import dev.vortex.api.Array;
 import dev.vortex.api.DType;
 import java.util.OptionalLong;
-
-@Open
-public class JNIArray implements Array {
+import org.apache.arrow.c.ArrowArray;
+import org.apache.arrow.c.ArrowSchema;
+import org.apache.arrow.c.CDataDictionaryProvider;
+import org.apache.arrow.c.Data;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.VectorSchemaRoot;
+
+public final class JNIArray implements Array {
     static {
         NativeLoader.loadJni();
     }
 
+    // TODO(aduffy): safe to assume this class doesn't need to be threadsafe?
```

Comment:
Hmm. Should we have a blanket notice somewhere that the classes of this library are not meant to be shared by multiple threads? The batch iterator below also seems thread unsafe, right? I guess that specific class is covered under the wider umbrella of Spark's threading model.